client/src/components/profile/components/time-line.tsx

```diff
@@ -265,15 +266,19 @@ function useIdToNameMap(t: TFunction): Map<string, NameMap> {
           // @ts-expect-error Graphql needs typing
           id,
           // @ts-expect-error Graphql needs typing
+          superBlock,
+          // @ts-expect-error Graphql needs typing
           title,
           // @ts-expect-error Graphql needs typing
           fields: { slug, blockName }
         }
       }
     }) => {
+      const blockNameTitle = t(`intro:${superBlock}.blocks.${blockName}.title`);
+      const step = t(`aria.step`);
       idToNameMap.set(id, {
         challengeTitle: `${
-          title.includes('Step') ? `${blockName} - ` : ''
+          title.includes(step) ? blockNameTitle + ' - ' : ''
```

Comment:
```suggestion
          hasEditableBoundaries ? blockNameTitle + ' - ' : ''
```
Checking for the `aria.step` in the title does work (and this is a good catch!). However, the presence of the word "Step" in the title is not necessary for the challenge to part of a practice project. We do that, by convention, but it's not enforced and future authors could use a different word or simply drop it entirely and just use the number.

`hasEditableBoundaries` is necessary, though, since all steps have embedded instructions.